File: backend/app/schemas/indicator.py
from pydantic import BaseModel
from datetime import datetime


class IndicatorPoint(BaseModel):
    timestamp: datetime
    value: float | None


class IndicatorOut(BaseModel):
    symbol: str
    interval: str
    type: str
    period: int
    data: list[IndicatorPoint]


class BollingerPoint(BaseModel):
    timestamp: datetime
    upper: float | None
    middle: float | None
    lower: float | None


class BollingerOut(BaseModel):
    symbol: str
    interval: str
    type: str = "bbands"
    period: int
    data: list[BollingerPoint]

File: frontend/src/types/indicator.ts
export interface IndicatorPoint {
  timestamp: string;
  value: number | null;
}

export interface BollingerPoint {
  timestamp: string;
  upper: number | null;
  middle: number | null;
  lower: number | null;
}

export interface IndicatorData {
  symbol: string;
  interval: string;
  type: string;
  period: number;
  data: IndicatorPoint[];
}
